### PR TITLE
Request for code review on Ruby Enumerable Methods 

### PR DESCRIPTION
**Pull request by Diegodsha**
In this project, we replicated some of Ruby's enumerable methods. We created :
* my_each to function exactly as Ruby's each method.
* my_each_with_index to function exactly as Ruby's each_with_index method.
* my_all? to function exactly as Ruby's all? method.
* my_any? to function exactly as Ruby's any? method.
* my_none? to function exactly as Ruby's none? method.
* my_select to function exactly as Ruby's select method.
* my_map to function exactly as Ruby's map method.
* my_inject to function exactly as Ruby's inject method.
* my_count to function exactly as Ruby's count method.

### CODE REVIEW

File: enumerable_module.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i]
+    end
+  end
```
Comment (jubaan) [inline enumerable_module.rb]: - [x] `my_each` is not working as the original `each` method, also is not passing all test cases.

### `each` is an Enumerator that has these characteristics: 
- it returns the array/range/hash itself if a block is given.

```
[1, 2, 3].each { |num| num }
# [1, 2, 3]
(1..3).each { |num| num * 2 }
# 1..3
````

[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerator.html#method-i-each)

File: enumerable_module.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i], i
+    end
+  end
```
Comment (jubaan) [inline enumerable_module.rb]: - [x] `my_each_with_index` is not working as the original `each_with_index` method, also is not passing all test cases.

### `each_with_index` is an Enumerable that has these characteristics: 
- it returns the array/range/hash itself if a block is given.

```
[1, 2, 3].each_with_index { |num, idx| num }
# [1, 2, 3]
(1..3).each { |num, idx| num * idx }
# 1..3
````

[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-each_with_index)

File: enumerable_module.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    elsif !param.nil? && !sym.nil?
+      to_a.my_each { |item| param = param.send(sym, item) }
+    elsif param.is_a?(Symbol) && sym.nil?
+      accum = to_a[0]
+      (to_a.length - 1).times { |index| accum = accum.send(param, to_a[index + 1]) }
+      param = accum
+    end
+    param
+  end
+end
```
Comment (jubaan) [inline enumerable_module.rb]: - [x] `my_inject` is not working as the original `inject` method, also is not passing all test cases.

### `inject`` is an Enumerable that has these characteristics:
- When a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block

[Enumerable#inject](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-inject)

**Review by jubaan — CHANGES_REQUESTED** [review]

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED!! :recycle: :red_circle: 
---------------------------
Hello team!! :wave: you are doing an amazing job so far :clap: :100: 
There are a few things to change :hammer_and_wrench: see the below :point_down: 
----------------------

- [ ] You have debugging code. All your `main.rb` file is debugging code. It's really nice that you want to show your methods working but isn't necessary, and it is debugging code. You souldn't commit debugging code. [HERE](https://questions.microverse.org/t/debugging-instructions-code/346) you can read the reasons WHY you shouldn't. You can keep that file for yourself on a different branch or put it inside `.gitignore`, if you want to keep it, but don't commit it.
- [ ] Don't forget to move your `multiply_els` method from `main.rb` to `enumerable_module.rb` since you are not going to commit this file.
---------------------------
Make the required changes :hammer_and_wrench: :muscle: and send it for another review :+1: 
HAPPY CODDING !!! 

MB5W4D

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉 

**Congratulations team 👏, and good luck with your next project 😃.**

**Pull request closed without merging**